.github/workflows/build.yml

```diff
@@ -86,6 +86,11 @@ jobs:
     needs: smoke
     runs-on: ubuntu-latest
     if: startsWith(github.ref, 'refs/tags/v')
+
+    permissions:
+      contents: write
+      packages: write
```

Comment:
These permissions are only specified to restrict the permission of the [GitHub Token created automatically](https://docs.github.com/en/actions/security-guides/automatic-token-authentication).
Do you prefer using your `PAT` secret? I noticed that you're using `${{ secrets.PAT }}` everywhere, instead of `${{ github.token }}`, but if you want to use the PAT for pushing the Docker image, it'll need the `write:packages` permission.